File: Notification-Service/src/main/java/byrd/riley/notificationservice/model/QueueReceiver.java
package byrd.riley.notificationservice.model;

import org.springframework.amqp.rabbit.annotation.RabbitHandler;
import org.springframework.amqp.rabbit.annotation.RabbitListener;

@RabbitListener(queues = {"task_queue"}, id = "mainListener")
public class QueueReceiver {

	private Texter texter;

	private Emailer emailer;
	
	public QueueReceiver(Texter texter) {
		this(texter, null);
	}
	
	public QueueReceiver(Emailer emailer) {
		this(null, emailer);
	}
	
	public QueueReceiver(Texter texter, Emailer emailer) {
		this.texter = texter;
		this.emailer = emailer;
	}
	
	
	@RabbitHandler
	public void receive(String in) {
		System.out.println(" [x] Received '" + in + "'");
		
		String[] messageContents = in.split("\r\n");
		String messageType = messageContents[0];
		String receiver = messageContents[1];
		String messageBody = messageContents[2];

    	if(messageType.equalsIgnoreCase("email"))
    		emailer.sendEmail(receiver, "My Email", messageBody);
    	else if(messageType.equalsIgnoreCase("text"))
    		texter.sendText(receiver, messageBody);

        System.out.println(" [x] Done");
	}
	
}

File: Notification-Service-Kotlin/src/main/kotlin/byrd/riley/notificationservicekotlin/model/QueueReceiver.kt
package byrd.riley.notificationservicekotlin.model

import org.springframework.amqp.rabbit.annotation.RabbitHandler
import org.springframework.amqp.rabbit.annotation.RabbitListener

@RabbitListener(queues = ["task_queue"], id = "mainListener")
class QueueReceiver(private val texter: Texter?, private val emailer: Emailer?) {

    constructor(texter: Texter?): this(texter, null)
    constructor(emailer: Emailer?): this(null, emailer)

    @RabbitHandler
    fun receive(`in`: String) {
        print(" [x] Received '${`in`}'")

        val messageContents: List<String> = `in`.split("\r\n")

        val messageType: String = messageContents[0]
        val receiver: String = messageContents[1]
        val messageBody: String = messageContents[2]

        if(messageType.equals("email", true))
            emailer?.sendEmail(receiver, "My Email", messageBody)
        else if(messageType.equals("text", true))
            texter?.sendText(receiver, messageBody)

        print(" [x] Done")
    }
}